libcomposefs/lcfs-writer-erofs.c

```diff
@@ -899,6 +934,24 @@ static int add_overlayfs_xattrs(struct lcfs_node_s *node)
 		}
 	}
 
+	/* escape whiteouts */
+	if (type == S_IFCHR && node->inode.st_rdev == makedev(0, 0)) {
+		struct lcfs_node_s *parent = lcfs_node_get_parent(node);
+
+		lcfs_node_set_mode(node,
+				   S_IFREG | (lcfs_node_get_mode(node) & ~S_IFMT));
+		ret = lcfs_node_set_xattr(node, "trusted.overlay.overlay.whiteout",
```

Comment:
Optional future cleanup: we could have a `linux-overlayfs-internals.h` header that includes `#define`s that link to the upstream source code.

Reply:
I might look at extracting this. Although we can't use the kernel headers here, the way the xattr defines are structured in the kernel is a bit weird.